src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's Contact details in the addressbook. Not meant to be
+ * used directly, but rather meant to be subclassed.
+ * 
+ * @author varungupta
+ */
+public class Contact {
+	protected String value;
+	protected boolean isPrivate;
+
```

Comment:
none of the subclasses are using the parent class's constructor, you should create a constructor for the parent class. If you're unsure about inheritance, there are many resources online that talk about this topic, such as this tutorial by Sun Oracle: https://docs.oracle.com/javase/tutorial/java/IandI/subclasses.html